cli/objref.go

```diff
@@ -67,3 +69,113 @@ func parseNestedObjectID(ctx context.Context, startingDir fs.Entry, parts []stri
 
 	return e.(object.HasObjectID).ObjectID(), nil
 }
+
+func findSnapshotsByRootObjectID(ctx context.Context, rep repo.Repository, rootID object.ID) ([]*snapshot.Manifest, error) {
```

Comment:
nit: Does it make sense to move these helpers to the `snapshot` package?
And then export `FindSnapshotsByRootObjectID`, `FindSnapshotByRootObjectIDOrManifestID` and `FilesystemEntryFromID`
Thoughts?

Reply:
done, this actually helped clean up the code in a whole bunch of places.